src/components/chip-group/chip-group.tsx

```diff
@@ -0,0 +1,197 @@
+  shadow: true
+})
+export class ChipGroup implements InteractiveComponent {
+  //--------------------------------------------------------------------------
+  //
+  //  Element
+  //
+  //--------------------------------------------------------------------------
+
+  @Element() el: HTMLCalciteChipGroupElement;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Public Properties
+  //
+  //--------------------------------------------------------------------------
+
+  /** When true, interaction is prevented and the component is displayed with lower opacity. */
+  @Prop({ reflect: true }) disabled = false;
+
+  /** Accessible name for the component. */
+  @Prop() label!: string;
+
+  /** Specifies the size of the component. */
+  @Prop({ reflect: true }) scale: Scale = "m";
+
+  /** Specifies the selection mode of the component. */
+  @Prop({ reflect: true }) selectionMode: Extract<
+    "multiple" | "single" | "single-persist" | "none",
+    SelectionMode
+  > = "none";
+
+  /**
+   * Specifies the component's selected items.
+   *
+   * @readonly
+   */
+  @Prop({ mutable: true }) selectedItems: HTMLCalciteChipElement[] = [];
+
+  @Watch("selectedItems")
+  onMessagesChange(): void {
+    this.calciteChipGroupSelectChange.emit();
+  }
+
+  //--------------------------------------------------------------------------
+  //
+  //  Private Properties
+  //
+  //--------------------------------------------------------------------------
+  mutationObserver = createObserver("mutation", () => this.updateItems);
+
+  /** created list of Chip items */
+  private items: HTMLCalciteChipElement[] = [];
+
+  //--------------------------------------------------------------------------
+  //
+  //  Events
+  //
+  //--------------------------------------------------------------------------
+
+  /**
+   *
+   * @internal
+   */
+  @Event({ cancelable: false })
+  calciteChipInternalSelectionChange: EventEmitter<HTMLCalciteChipElement>;
+
+  /**
+   * Emits when the component's selection changes.
+   */
+  @Event({ cancelable: false }) calciteChipGroupSelectChange: EventEmitter<void>;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Lifecycle
+  //
+  //--------------------------------------------------------------------------
+  connectedCallback(): void {
+    this.mutationObserver?.observe(this.el, { childList: true, subtree: true });
+    this.updateItems;
+  }
+
+  componentDidRender(): void {
+    updateHostInteraction(this);
+  }
+
+  disconnectedCallback(): void {
+    this.mutationObserver?.disconnect();
+  }
+  //--------------------------------------------------------------------------
+  //
+  //  Event Listeners
+  //
+  //--------------------------------------------------------------------------
+
+  @Listen("calciteInternalChipKeyEvent")
+  calciteInternalChipKeyEvent(event: CustomEvent): void {
+    if (this.el === event.detail.target.parentElement) {
```

Comment:
It might be better if it were something like...

```
if(event.composedPath().contains(this.el){
```

So that its not relying on the parent structure.